src/pipelines/stage.ts

```diff
@@ -43,29 +46,34 @@ export interface AlarmProps {
 
 export abstract class DataStage extends Stage {
   readonly cloudwatchAlarms: cloudwatch.Alarm[];
+  readonly alarmsEnabled?: boolean;
 
   constructor(scope: Construct, id: string, props: DataStageProps) {
     super(scope, id, props);
 
+    this.alarmsEnabled = props.alarmsEnabled === undefined ? true : props.alarmsEnabled;
```

Comment:
You could rewrite this as:
```javascript
this.alarmsEnabled = props.alarmsEnabled ?? true;
```

Also, you don't need to make `alarmsEnabled` optional. You can change the definition to:
```javascript
readonly alarmsEnabled: boolean;
```

Reply:
Changed it to required, but how do I go about handling allowing `alarmsEnabled` to be optional in classes that inherit from `DataStage`? I don’t want this to be a mandatory parameter for all stages and would like it to default to true